core/hash/comparison_spec.rb

```diff
@@ -0,0 +1,97 @@
+require File.expand_path('../../../spec_helper', __FILE__)
+
+ruby_version_is "2.3" do
+  describe "Hash#<=" do
```

Comment:
Technically, this should be in `core/hash/lte_spec.rb`, following how it is done in `core/module`.
Although I do like having all examples about similar functionality in the same file too, I think it's better to be consistent in this case.